### PR TITLE
modemmanager: 1.14.2 -> 1.16.6

### PR DESCRIPTION
**Pull request by alyssais**
Getting the new test to work was an absolute nightmare, but I did it!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127235` at 3c85df57 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>45 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/bfdec977e16eda03caef298701ecd82c">appgate-sdp</a></li>
    <li><a href="https://gist.github.com/4dd2c25f763f7c3e5ac563fbd0fec7aa">clight</a></li>
    <li><a href="https://gist.github.com/5e79a04897298489c343aaa91f799d6b">flatpak</a></li>
    <li><a href="https://gist.github.com/baf27f22eb123ea53bb7d1bd88596ef2">gammastep</a></li>
    <li><a href="https://gist.github.com/8dbddbeb687cac1dd05d1d94371f0b44">geoclue2</a></li>
    <li><a href="https://gist.github.com/ba9115b8f97213b560705bf52638ff78">glib-networking</a> (gnome.glib-networking ,gnome.glib_networking)</li>
    <li><a href="https://gist.github.com/c6b6970714fab9734a1ddd6435077ca4">gnome-network-displays</a></li>
    <li><a href="https://gist.github.com/4b6dae1d184e27f116473f92627ab7dc">gnome.gnome-clocks</a></li>
    <li><a href="https://gist.github.com/de460fbbd980fe805c90d912365bbaac">gnome.gnome-settings-daemon</a> (gnome.gnome_settings_daemon)</li>
    <li><a href="https://gist.github.com/698f2a0bd943bbdc162cd85011648ee7">gnome.gnome-weather</a></li>
    <li><a href="https://gist.github.com/bf0c2dbd331d27789fdc9a5bcbfe38c0">gnome.mutter334</a> (pantheon.mutter)</li>
    <li><a href="https://gist.github.com/eff42b129af7e3fdab2e63eeb74735a5">networkmanager-fortisslvpn</a> (gnome.networkmanager-fortisslvpn)</li>
    <li><a href="https://gist.github.com/5cb03b7bf28195e85445f3fc14b84df2">networkmanager-iodine</a> (gnome.networkmanager-iodine)</li>
    <li><a href="https://gist.github.com/84bf3a8e4e1fb89fec8a565291dd0258">networkmanager-openconnect</a> (gnome.networkmanager-openconnect)</li>
    <li><a href="https://gist.github.com/265c0223eba5a07720c4d394fba53a8a">networkmanager-vpnc</a> (gnome.networkmanager-vpnc)</li>
    <li><a href="https://gist.github.com/27e9a272aa1c9471b57e902eeef31eac">networkmanagerapplet</a> (gnome.networkmanagerapplet)</li>
    <li><a href="https://gist.github.com/d03eec3cf958bc7b449c3050c193affc">libnma</a></li>
    <li><a href="https://gist.github.com/779cc810cd1cad294a33f113d4449a33">libproxy</a></li>
    <li><a href="https://gist.github.com/893cbc01c9bf94c8b3f527ab46fee8cc">libsForQt5.kdelibs4support</a> (libsForQt515.kdelibs4support ,plasma5Packages.kdelibs4support)</li>
    <li><a href="https://gist.github.com/1553e0b90abb29214df017f481c7df5e">libsForQt5.networkmanager-qt</a> (libsForQt515.networkmanager-qt ,plasma5Packages.networkmanager-qt)</li>
    <li><a href="https://gist.github.com/3a507c6ea36bb01f9092b10f82a72b9e">libsForQt514.kdelibs4support</a></li>
    <li><a href="https://gist.github.com/c12464549af55dbb07644109e103c3ab">modemmanager</a></li>
    <li><a href="https://gist.github.com/351bd6dca646eb06b2a2170492270ebe">networkmanager</a></li>
    <li><a href="https://gist.github.com/0afa64ff46f28e6d398ed18c110547e7">networkmanager_dmenu</a></li>
    <li><a href="https://gist.github.com/6c8191b99ae996608e59eda2fedaff83">ostree</a></li>
    <li><a href="https://gist.github.com/3faee6d03dd0b32caa6c8e9a4965c354">pantheon.elementary-dpms-helper</a></li>
    <li><a href="https://gist.github.com/27909757bb286975ab2b73337be3996f">pantheon.elementary-settings-daemon</a></li>
    <li><a href="https://gist.github.com/95e9c5465e40dad72e9142fa8dcda95c">pantheon.gala</a></li>
    <li><a href="https://gist.github.com/3f25abb7a5f83b70d5b0a25b932af7f3">pantheon.pantheon-agent-geoclue2</a></li>
    <li><a href="https://gist.github.com/8c9a7cf2ab9f5055c967f0d7422b9e4c">pantheon.switchboard-plug-mouse-touchpad</a></li>
    <li><a href="https://gist.github.com/b92e4c382bd317cbc1158b3d06c9b1db">pantheon.switchboard-plug-network</a></li>
    <li><a href="https://gist.github.com/17bad5665b59f669563672d59447c28d">pantheon.switchboard-plug-pantheon-shell</a></li>
    <li><a href="https://gist.github.com/aef3d98cb7c38d82486062aab68ae219">pantheon.switchboard-plug-power</a></li>
    <li><a href="https://gist.github.com/98115562151c2bfa1f2eefb83f4cf7a6">pantheon.wingpanel</a></li>
    <li><a href="https://gist.github.com/79c6b119261e3580c5af4156ac306a50">pantheon.wingpanel-indicator-bluetooth</a></li>
    <li><a href="https://gist.github.com/e4cd2910caf6a71ee7cd6f552836df37">pantheon.wingpanel-indicator-keyboard</a></li>
    <li><a href="https://gist.github.com/34be50be130ca9aaa003ad385fed1d2c">pantheon.wingpanel-indicator-network</a></li>
    <li><a href="https://gist.github.com/9d26824b6dbb96e6838d388128edf5dd">pantheon.wingpanel-indicator-nightlight</a></li>
    <li><a href="https://gist.github.com/5c7b6bafa4f6432b05cf7a0ae263fcf5">pantheon.wingpanel-indicator-notifications</a></li>
    <li><a href="https://gist.github.com/92229f12874e76e5bebcc8dd0b3cf31e">pantheon.wingpanel-indicator-power</a></li>
    <li><a href="https://gist.github.com/6e7ad0bc0e9c7620148f380b4147d3ea">pantheon.wingpanel-indicator-session</a></li>
    <li><a href="https://gist.github.com/a8615dbb45a1483a77d6218198c2f4b6">pantheon.wingpanel-indicator-sound</a></li>
    <li><a href="https://gist.github.com/d0c9230ac10f8a33347218b0157f6a0a">python38Packages.python-dbusmock</a></li>
    <li><a href="https://gist.github.com/57ca2c930357f6ca8bbeb47b76f8c4c2">redshift</a></li>
    <li><a href="https://gist.github.com/bccc86a371cc9679221cce6169ddf4d8">redshift-wlr</a></li>
  </ul>
</details>
<details>
  <summary>339 packages skipped due to time constraints:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>almanah</li>
    <li>apostrophe</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>astroid</li>
    <li>balsa</li>
    <li>bibata-cursors</li>
    <li>...</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by maxeaubrey** [comment]
Looks like modemmanager needs `libqmi` to get bumped to 1.28.6. Thankfully that's just an easy bump and `modemmanager` builds fine for me after that.

Also thanks for fixing that test, I uhhh quickly lost interest when I looked at doing it myself :sweat_smile: 

(minor nit, it's 1.14.12, not 1.14.2)

**Comment by alyssais** [comment]
Well there goes my hope of this being borderline enough to skip staging...

libqmi bump is https://github.com/NixOS/nixpkgs/pull/127240.

And I'll fix the commit message -- thanks for pointing that out. :)

File: pkgs/tools/networking/modem-manager/default.nix [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
+
   nativeBuildInputs = [ vala gobject-introspection gettext pkg-config ];
 
   buildInputs = [ glib udev libgudev polkit ppp libmbim libqmi systemd ];
 
+  installCheckInputs = [
+    python3 python3.pkgs.dbus-python python3.pkgs.pygobject3
+  ];
+
+  postPatch = ''
```
Comment (jtojnar) [inline pkgs/tools/networking/modem-manager/default.nix:32]: Could you please move this next to the other phase hooks as per https://discourse.nixos.org/t/document-attribute-ordering-in-package-expressions/4887?

**Review by jtojnar — COMMENTED** [review]
Changes look okay:

- 1.16 https://lists.freedesktop.org/archives/modemmanager-devel/2021-February/008437.html

  - > A new 'qcom-soc' plugin is implemented to be able to use ModemManager in Qualcomm SoCs like the MSM8916 or MSM8974. This plugin uses a combination of RPMSG based control ports plus BAM-DMUX based network ports. This plugin is disabled by default, even when `--enable-all-plugins` is used, and if wanted it must be explicitly enabled with `--enable-plugin-qcom-soc`. Systems targeting this kind of SoCs, like postmarketos, should enable it. Standard distributions may or may not include it, up to the targeted hardware in each distribution.

    We can probably enable it when someone asks for it.

- 1.16.2 https://lists.freedesktop.org/archives/modemmanager-devel/2021-March/008460.html
- 1.16.4 https://lists.freedesktop.org/archives/modemmanager-devel/2021-April/008506.html
- 1.16.6 https://lists.freedesktop.org/archives/modemmanager-devel/2021-June/008660.html
  - aforementioned libqmi bump requirement


On unrelated note, we should probably add the freedesktop team to maintainers here as well.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**